### DISCUSSION TITLE
Workspaces - Best Practices

### DISCUSSION THREAD

**Discussion by jinsley8** [hoppscotch/hoppscotch#3725] [Q&A] [answered]
Is it best practice to use one workspace per project or use a workspace as a team with multiple projects (collections). 

I've setup a workspace for my team with multiple project collections with endpoint subfolders but I run into issues with the environment variables being somewhat redundant.

I may have `baseURL` but it's different for each project so I have to make `baseURL_ABC`, `baseURL_DEF`, `baseURL_GHI`, etc. 

How is everyone else handling this?

Is the a restriction to maximum number of workspaces created?

**Comment by liyasthomas** [comment, marked as answer]
If you've multiple projects within a workspace you can create a team environment dedicated to each project with different values to the same variable `baseURL`. This, along with the proper use of collections should be able to deliver your workflow.

**Reply by jinsley8** [reply]
Instead of having many Teams (Workspaces) now I'd have many environments.

Right now I have the following 3 Environments where something like `baseURL_ABC` and `baseURL_DEF` is different on each one:
- Dev
- Staging
- Prod

I think what you're saying is to add Environments for each project and `baseURL` for all:
- Dev_ABC
- Staging_ABC
- Prod_ABC

- Dev_DEF
- Staging_DEF
- Prod_DEF

It's sort of the same thing. You either have a big list of Environments or variables. I guess I'll just keep doing it the original way for now as it's easier to manage the variables.

Current Workflow:
- I have 3 Workspaces (2 companies and personal), each workspace has a Collection per project or API.
- I have those 3 environments and then project-based variables for each environment.


Thanks for the advice!
